packages/ihe-gateway/server/CodeTemplates/XCA Library - Outbound/Process RegistryObjectList response/Process RegistryObjectList response.js

```diff
@@ -54,7 +54,24 @@ function processRegistryObjectList(list) {
 	
 				var created = entry.*::Slot.("creationTime" == @name).*::ValueList.*::Value.toString();
 				if (created) try {
-					attachment.creation = DateUtil.convertDate('yyyyMMddhhmmss', "yyyy-MM-dd'T'hh:mm:ss", created);
+					var timestampLength = created.length;
+					if (timestampLength === 8) {
+						attachment.creation = DateUtil.convertDate('yyyyMMdd', "yyyy-MM-dd", created);
+					}
+					if (timestampLength === 14) {
+						attachment.creation = DateUtil.convertDate('yyyyMMddhhmmss', "yyyy-MM-dd'T'hh:mm:ss", created);
+					}
+				} catch(ex) {}
+
+				var serviced = entry.*::Slot.("serviceStartTime" == @name).*::ValueList.*::Value.toString();
+				if (serviced) try {
```

Comment:
Is this an instance exposing us to this bug/unexpected behavior again? https://github.com/metriport/metriport/pull/1964/files

Shouldnt we use if (serviced.length > 0) and ditto for created? Like have we tested the behavior where we do have serviceStartTime.ValueList but no Value? Ditto for above at "created"

Reply:
Good point. Will test that and add the length check if needed

Reply:
Just tested it and it works fine here. This is likely because we're using `.toString()` first. After that, the if-check works as expected.